storage/src/test/scala/coop/rchain/storage/TestTools.scala

```diff
@@ -46,19 +50,27 @@ object TestTools {
     blobs
   }
 
-  def ArraysEqual(a1: Array[String], a2: Array[String]): Boolean = {
-    if (a1.length != a2.length) return false
-    for (i <- 0 until a1.length) {
-      var contains = false
-      for (j <- 0 until a2.length) {
-        if (a1(i) == a2(j)) {
-          contains = true
+  def arraysEqual(a1: Array[String], a2: Array[String]): Boolean = {
```

Comment:
The current function gives the following maybe-surprising result (two arrays with equal length and same elements in different proportion):
```Scala
scala> arraysEqual(Array("foo", "foo", "bar"), Array("foo", "bar", "bar"))
res16: Boolean = true
```
If this is really what you want, I'd just use the Scala `Set` business for clarity:
```Scala
def arraysEqual(a1: Array[String], a2: Array[String]): Boolean =
  a1.length == a2.length && a1.toSet == a2.toSet
```
If you want to compare two multisets, then maybe just sort some seqs and compare those?
```Scala
def arraysEqual(a1: Array[String], a2: Array[String]): Boolean =
  a1.toSeq.sorted == a2.toSeq.sorted
```
or
```Scala
def arraysEqual(a1: Array[String], a2: Array[String]): Boolean =
  a1.sorted.sameElements(a2.sorted)
```

Reply:
Fixed